File: 01-ordenacao/csharp/OrdenacaoMVC/src/Model/Algoritmos/IAlgoritmoOrdenacao.cs
namespace OrdenacaoMVC.Model.Algoritmos;

/// <summary>
/// Contrato comum aos algoritmos de ordenação. Permite que o Controller
/// trate diferentes algoritmos de forma polimórfica
///</summary>
public interface IAlgoritmoOrdenacao
{
    /// <summary>
    /// Nome exibido no menu da aplicação
    ///</summary>
    string Nome { get; }

    /// <summary>
    /// Ordena a lista in-place e retorna as métricas da execução
    ///</summary>
    ResultadoOrdenacao Ordenar(List<int> lista);
}

File: 01-ordenacao/java/OrdenacaoMVC/src/model/algoritmos/IAlgoritmoOrdenacao.java
package model.algoritmos;

import java.util.List;
import model.ResultadoOrdenacao;

/**
 * Contrato dos algoritmos de ordenação
 */
public interface IAlgoritmoOrdenacao {

    /**
    * Nome exibido no menu 
    */
    String getNome();

    /**
     * Executa o algoritmo de ordenação na lista fornecida
     * @param lista lista de números a ordenar
     * @return ResultadoOrdenacao métricas da execução
     */
    ResultadoOrdenacao ordenar(List<Integer> lista);
}
